libraries/Microsoft.Bot.Builder.LanguageGeneration/Extensions.cs

```diff
@@ -160,5 +161,23 @@ public static string GetPrefixErrorMessage(this LGFileParser.NormalTemplateStrin
 
             return errorPrefix;
         }
+
+        /// <summary>
+        /// Convert antlr parser into Range.
+        /// </summary>
+        /// <param name="context">antlr parse context.</param>
+        /// <param name="lineOffset">line offset.</param>
+        /// <returns>Range object.</returns>
+        public static Range ConvertToRange(this ParserRuleContext context, int lineOffset = 0)
+        {
+            if (context == null)
+            {
+                return Range.DefaultRange;
+            }
+
+            var startPosition = new Position(lineOffset + context.Start.Line, context.Start.Column);
+            var stopPosition = new Position(lineOffset + context.Stop.Line, context.Stop.Column + context.Stop.Text.Length);
```

Comment:
Is this about getting file specific offsets rather than merged offsets when reporting errors?

Reply:
the error line number of specific template is related to the current template, not to the whole LG file, so, we added a offset of the template start line.